object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionTransfer.java

```diff
@@ -74,9 +88,54 @@ public ObjectConstructionTransfer(final CFAnalysis analysis) {
             result.getElseStore(),
             exceptionalStores);
     exceptionalStores = null;
+
+    Node receiver = node.getTarget().getReceiver();
+    addTemporaryVars(result, node);
+    if (atypefactory.biMap.inverse().containsKey(receiver)) {
+      String methodName = node.getTarget().getMethod().getSimpleName().toString();
+      methodName =
+          ((CalledMethodsAnnotatedTypeFactory) atypeFactory)
+              .adjustMethodNameUsingValueChecker(methodName, node.getTree());
+      accumulate(atypefactory.biMap.inverse().get(receiver), result, methodName);
+    }
+
     return finalResult;
   }
 
+  @Override
+  public TransferResult<CFValue, CFStore> visitObjectCreation(
+      ObjectCreationNode node, TransferInput<CFValue, CFStore> input) {
+    TransferResult<CFValue, CFStore> result = super.visitObjectCreation(node, input);
+    addTemporaryVars(result, node);
+    return result;
+  }
+
+  private void addTemporaryVars(TransferResult<CFValue, CFStore> result, Node node) {
```

Comment:
Please add Javadoc

Reply:
Also this method needs a better name.  Maybe `updateStoreWithTempVar`?  I think the method either creates or looks up the temp var `t` for `node`, and then updates the store to give `t` the same type as `node`.  Is that correct?

Reply:
That's correct, I updated the name and added Javadoc:
https://github.com/kelloggm/object-construction-checker/pull/291/files#diff-3fc5dcd77eafbbf61a8da8156fac14c917b05d258d829aab8e8706b815b91b52R137